app/main.py

```diff
@@ -2,0 +1,32 @@
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.health = health
+        self.name = name
+        self.hidden = False
+
+        self.alive.append(self)
+
+    def die(self) -> None:
+        if self.health <= 0:
+            Animal.alive.remove(self)
+
+    def __repr__(self) -> str:
+        string = "{"
+        string += (f"Name: {self.name}, "
+                   f"Health: {self.health}, Hidden: {self.hidden}")
+        string += "}"
+        return string
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, victim: Herbivore) -> None:
+        if isinstance(victim, Herbivore) and not victim.hidden:
```

Comment:
The check `isinstance(victim, Herbivore)` is redundant because the method signature already specifies that `victim` should be a `Herbivore`. Consider removing this check unless you anticipate calling `bite` with other types in the future.